EIPS/eip-3155.md

```diff
@@ -0,0 +1,241 @@
+```
+
+### Summary and error handling
+
+At the end of execution, the CUT MUST print some summerical info, this info SHOULD have the following fields. The summary SHOULD be printed to STDOUT.
+
+Required Fields: 
+- `stateRoot` Root of the state trie after executing the transaction
+- `output` Return values of the function
+- `gasUsed` All gas used by the transaction
+- `successful` Bool whether transaction was executed successfully
+
+OptionalFields:
+- `time` Time needed to execute the transaction
+```
+{"stateRoot":"0xd4c577737f5d20207d338c360c42d3af78de54812720e3339f7b27293ef195b7","output":"","gasUsed":"0x3","successful":"true","time":141485}
+```
+
+The state test format allows for multiple `postStates` for different forks.
+If different `postStates` are present the CUT MUST execute the test according to the rules of the fork sequentially.
+It MUST print a trace for each execution as well as an individual summary for each test.
+
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+This EIP is created to enable easier testing of EVM implementations to create a safer ecosystem. It is largely based on the previous non-official documentation for EVM tracing. It tries to cover as many corner cases as possible to enable true client compatibility.
+This proposal is heavily influenced by the requirements for [goevmlab](https://github.com/holiman/goevmlab) and our effort to create differential fuzzers for the EVM.
+
+## Backwards Compatibility
+<!--All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+This EIP is fully backward compatible with ethereum as it only introduces a better tracing infrastructure that is optional for clients to implement.
+
+### Clients
+This EIP is fully backward compatible with go-ethereum. OpenEthereum, Besu and Nethermind clients would have to change their JSON output of `openethereum-evm` `evmtool` and `nethtest` slightly do adhere to the new and stricter specs. New clients would need to implement this change if they want to be part of the differential fuzzing group.
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Other EIPs can choose to include links to test cases if applicable.-->
+<details><summary>Test cases</summary>
```

Comment:
There should be tests exercising `memSize`, `refundCounter`, `returnStack`, and `returnData` since they were called out in the prologue.   Referencing a reference test and including the relevant line would satisfy me.